packages/core/src/renderer/components/tree-view/tree-view.tsx

```diff
@@ -3,103 +3,131 @@
-    event.preventDefault();
-    scrollToItem(nodeId);
-  };
+  return (
+    <li
+      className={cssNames(props.classes?.root, optionalCssNames, styles.treeItem, {
+        [styles.selected]: props.selected ?? false,
+      })}
+      role="treeitem"
+      data-testid={props["data-testid"]}
+      onClick={props.onClick}
+      onMouseOver={() => setHovering(true)}
+      onMouseLeave={() => setHovering(false)}
+    >
+      <div className={cssNames(props.classes?.iconContainer, styles.iconContainer)}>
+        {props.icon}
+      </div>
+      <div className={props.classes?.label}>
+        {props.label}
+      </div>
+    </li>
+  );
+}
 
-  const renderTree = (nodes: NavigationTree[]) => {
-    return nodes.map(node => (
-      <TreeItem
-        key={node.id}
-        nodeId={node.id}
-        label={node.name}
-        onLabelClick={(event) => onLabelClick(event, node.id)}
-        className={cssNames({ selected: node.selected })}
-        title={node.name}
-      >
-        {Array.isArray(node.children) ? node.children.map((node) => renderTree([node])) : null}
-      </TreeItem>
-    ));
-  };
+export interface TreeGroupClasses {
+  root?: string;
+  group?: string;
+  iconContainer?: string;
+  label?: string;
+  contents?: string;
+}
 
-  useEffect(() => {
-    if (!prevData.current.length) {
-      expandTopLevelNodes();
-    } else {
-      expandParentNode();
-    }
-    prevData.current = data;
-  }, [data]);
+export interface TreeGroupProps {
+  classes?: TreeGroupClasses;
+  children?: JSX.Element[] | JSX.Element;
+  defaultExpanded?: boolean;
+  label: JSX.Element | string;
+  "data-testid"?: string;
+  collapseIcon?: JSX.Element;
+  expandIcon?: JSX.Element;
+}
 
-  if (!data.length) {
-    return null;
-  }
+export function TreeGroup(props: TreeGroupProps) {
+  const [expanded, setExpanded] = useState(props.defaultExpanded ?? true);
 
   return (
-    <TreeView
-      data-testid="TreeView"
-      className="TreeView"
-      expanded={expanded}
-      onNodeToggle={handleToggle}
-      defaultCollapseIcon={<Icon material="expand_more"/>}
-      defaultExpandIcon={<Icon material="chevron_right" />}
+    <li
+      className={cssNames(props.classes?.root, styles.treeGroup)}
+      role="group"
+      data-testid={props["data-testid"]}
     >
-      {renderTree(data)}
-    </TreeView>
+      <div
+        className={cssNames(props.classes?.group, styles.group)}
+        onClick={() => setExpanded(!expanded)}
+      >
+        <div className={cssNames(props.classes?.iconContainer, styles.iconContainer)}>
+          {
+            expanded
+              ? props.collapseIcon ?? <Icon material="expand_more" />
+              : props.expandIcon ?? <Icon material="chevron_right" />
+          }
+        </div>
+        <div className={props.classes?.label}>
+          {props.label}
+        </div>
+      </div>
+      <ul
+        className={cssNames(props.classes?.contents, styles.contents, {
```

Comment:
`props.classes?` is repeated 6 times here. Recommend you to extract it in constructor:

```tsx
export function TreeGroup({classes, children, ...props}: TreeGroupProps)
```